SU2_CFD/src/solvers/CNEMONSSolver.cpp

```diff
@@ -285,102 +265,115 @@ void CNEMONSSolver::BC_HeatFluxNonCatalytic_Wall(CGeometry *geometry,
                                                  unsigned short val_marker) {
 
   /*--- Local variables ---*/
-  bool implicit;
-  unsigned short iDim, iVar;
-  unsigned short T_INDEX, TVE_INDEX, RHOCVTR_INDEX;
-  unsigned long iVertex, iPoint, total_index;
-  su2double dTdn, dTvedn, ktr, kve, pcontrol, Wall_HeatFlux;
-  su2double *Normal, Area, *V;
-
-  /*--- Assign booleans ---*/
-  implicit = (config->GetKind_TimeIntScheme_Flow() == EULER_IMPLICIT);
+  const bool implicit = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT);
+  const auto Marker_Tag = config->GetMarker_All_TagBound(val_marker);
+  su2double Wall_HeatFlux = config->GetWall_HeatFlux(Marker_Tag)/config->GetHeat_Flux_Ref();
 
   /*--- Set "Proportional control" coefficient ---*/
-  pcontrol = 1.0;
-
-  /*--- Identify the boundary by string name ---*/
-  string Marker_Tag = config->GetMarker_All_TagBound(val_marker);
-
-  /*--- Get the specified wall heat flux from config ---*/
-  Wall_HeatFlux = config->GetWall_HeatFlux(Marker_Tag);
+  su2double pcontrol = 1.0;
 
   /*--- Get the locations of the primitive variables ---*/
-  T_INDEX       = nodes->GetTIndex();
-  TVE_INDEX     = nodes->GetTveIndex();
-  RHOCVTR_INDEX = nodes->GetRhoCvtrIndex();
+  unsigned short T_INDEX       = nodes->GetTIndex();
+  unsigned short TVE_INDEX     = nodes->GetTveIndex();
+  unsigned short RHO_INDEX     = nodes->GetRhoIndex();
+  unsigned short RHOCVTR_INDEX = nodes->GetRhoCvtrIndex();
```

Comment:
constify?

Reply:
Yup.  I did all the ones I could find in the NEMONSSolver.  Needs to be done elsewhere.